recipes/newrelic/apm/dotNet/windows/net-install.yml

```diff
@@ -81,7 +81,9 @@ install:
           $xdoc = New-Object System.Xml.XmlDocument;
           $xdoc.PreserveWhitespace = $true;
           $xdoc.Load($file);
-          $xdoc.configuration.application.RemoveAll();
+          if ($xdoc.GetElementsByTagName("application")[0].HasChildNodes){
```

Comment:
should we check the call to GetElementsByTagName( ) returns at least 1 element, to avoid having another (index out of range) exception?

Reply:
Nice catch! Let me fix that.

Reply:
After looking it over and running some tests, I don't think we need to make any changes.
- The `application` element is a required element so it should never be removed and will prevent the agent from starting if it is.
- I did some testing and `$xdoc.GetElementsByTagName("application")[0].HasChildNodes` does not throw an error if the element does not exist.  It just returns nothing which powershell treats as false.

Reply:
We can leave it like that, but the moment `application` is no longer a required element (schema change), this will blow up. Perhaps we'd want it to be?

Reply:
It should fail gracefully and not error out, but yeah it will stop doing its job.